slides/src/content/services.md

```diff
@@ -114,4 +114,134 @@ import { HttpModule } from '@angular/http';
+          new Error(`${ e.status } ${ e.statusText }`)
+        );
+      }
+  });
+```
+
+---
+
+## Cancel a Request
+
+- One of the greatest benefits to using observables over promises is the ability to cancel `http` requests
+
+```ts
+  search() {
+    const request = this.searchService.search(this.searchField.value)
+      .subscribe(
+        result => { this.result = result.artists.items; },
+        err => { this.errorMessage = err.message; },
+        () => { console.log('Completed'); }
+      );
+
+    setTimeout(() => {request.unsubscribe()}, 0); // cancel request after 0 milliseconds
+  }
+```
+[View Example](http://plnkr.co/edit/XQL8v9?p=preview)
+
+---
+
+## Cancel a Request with `takeUntil`
+
+- Managing subscription cancellation via `unsubscribe` can get messy
+- The `takeUntil` operator allows us to compose your subscription management
+- `takeUntil` emits items from an observable until a given observable emits an item
+
+```ts
+search() {
+  const request = this.searchService.search(this.searchField.value)
+    .takeUntil(someOtherStream) // take until someOtherStream emits an item
+    .subscribe(
+      result => { this.result = result.artists.items; },
+      err => { this.errorMessage = err.message; },
+      () => { this.cancelMessage = 'Your request has been cancelled' }
+    );
+}
+```
+
+- **NOTE:** `takeUntil` will complete the observable unlike `unsubscribe`
+
+[View Example](https://plnkr.co/edit/v2aGkTCmi34jBr7an1i8?p=preview)
+
+---
+
+## Retry a request
+
+- It is possible to retry a failed request using the `.retry` operator when something goes wrong, for example, the user's connection may dropped.
+- `retry` takes a `retryCount` argument that specifies the number of times to retry a failed request.
+- If you do not specify `retryCount`, the request will be retried indefinitely.
+
+```ts
+  search(term: string) {
+    let tryCount = 0;
+    return this.http.get('https://api.spotify.com/v1/dsds?q=' + term + '&type=artist')
+      .map(response => response.json())
+      .retry(3);  // Will retry failed request 3 times
+}
+```
+- **Note:** The `onError` callback will not execute during the retry phase. The stream will only throw an error after the retry phase is complete
+
+[View Example](http://plnkr.co/edit/zSAWwV?p=preview)
+
+---
+
+## Converting requests into promises
+
+Using `toPromise()`, we can convert observables returned from Angular's http into promises
+
+```ts
+  search(term: string) {
+    return this.http
+      .get(`https://api.spotify.com/v1/search?q=${term}&type=artist`)
+      .map((response) => response.json())
+      .toPromise();
+  }
+```
+It could then be consumed as a regular promise
+
+```ts
+  search(someSearchString)
+    .then((result) => {
+      this.result = result.artists.items;
+    })
+    .catch((error) => console.error(error));
+```
+
+**Note:** Once converted into a promise you will lose the ability to cancel the request and chain RxJs operators
+
+---
```

Comment:
This last slide separator is not needed. It's creating an empty slide before the next section